gax-java/gax/src/main/java/com/google/api/gax/rpc/ClientContext.java

```diff
@@ -322,6 +325,9 @@ private static Map<String, String> getHeadersFromSettings(StubSettings settings)
     effectiveHeaders.putAll(userHeaders);
     effectiveHeaders.putAll(conflictResolution);
 
+    effectiveHeaders.computeIfPresent(
```

Comment:
I think handwritten libraries and/or customers can override credentials on request level, can we do more investigation? Maybe we can add this header on request level not client level.

Reply:
Let me look into this.

Reply:
Update on this: Consider this is a relatively niche case, to unblock this current change, I will look into this separately and create a followup if needed. Updated description to reflect.